src/fft/fftn.rs

```diff
@@ -0,0 +1,534 @@
+use mlx_macros::default_device;
+
+use crate::{array::Array, error::FftError, stream::StreamOrDevice};
+
+use super::{
+    resolve_size_and_axis_unchecked, resolve_sizes_and_axes_unchecked, try_resolve_size_and_axis,
+    try_resolve_sizes_and_axes,
+};
+
+/// One dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///   with zeros to match `n`. The default value is `a.shape[axis]`.
+/// - `axis`: Axis along which to perform the FFT. The default is -1.
+#[default_device(device = "cpu")]
+pub unsafe fn fft_device_unchecked(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Array {
+    let (n, axis) = resolve_size_and_axis_unchecked(a, n, axis);
+    unsafe {
+        let c_array = mlx_sys::mlx_fft_fft(a.c_array, n, axis, stream.stream.c_stream);
+        Array::from_ptr(c_array)
+    }
+}
+
+/// One dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
```

Comment:
for some you call this `n` and for some `s`

Reply:
This follows the python api. 1D fft uses n, n-D uses s